src/julia_internal.h

```diff
@@ -221,21 +222,25 @@ STATIC_INLINE int JL_CONST_FUNC jl_gc_szclass(size_t sz)
 #else
 #  define jl_is_constexpr(e) (0)
 #endif
-#define JL_SMALL_BYTE_ALIGNMENT 16
-#define JL_CACHE_BYTE_ALIGNMENT 64
-#define GC_MAX_SZCLASS (2032-sizeof(void*))
+
+#define jl_buff_tag ((uintptr_t)0x4eade800)
 
 STATIC_INLINE jl_value_t *jl_gc_alloc_(jl_ptls_t ptls, size_t sz, void *ty)
 {
     const size_t allocsz = sz + sizeof(jl_taggedvalue_t);
     if (allocsz < sz) // overflow in adding offs, size was "negative"
         jl_throw(jl_memory_exception);
+    size_t alignment = JL_SMALL_BYTE_ALIGNMENT;
+    if (ty && ((uintptr_t)ty != jl_buff_tag) &&
```

Comment:
Why? I need access to the alignment at this point and `ty` is used in `jl_set_typeof(v, ty)` later in the function.
Otherwise I will need to pass in the alignment from the outside.